src/main/scala/cognite/spark/SdkV1Rdd.scala

```diff
@@ -1,21 +1,26 @@
 package cognite.spark
 
-import cats.effect.IO
+import java.util.concurrent.{ArrayBlockingQueue, Executors}
+
+import cats.effect.{ContextShift, IO}
 import com.cognite.sdk.scala.v1.GenericClient
 import com.cognite.sdk.scala.common.Auth
 import org.apache.spark.{Partition, SparkContext, TaskContext}
 import org.apache.spark.rdd.RDD
 import org.apache.spark.sql.Row
-import fs2.Stream
+import fs2._
+
+import scala.collection.mutable
+import scala.concurrent.{ExecutionContext, Future}
 
 case class CdfPartition(index: Int) extends Partition
 
 case class SdkV1Rdd[A](
```

Comment:
Or we make the `getIOs` interface be "give me IOs for each partition" and let `getIOs` do merging as it pleases, and have number of partitions be = length of `getIOs` return value.

Would that work?

Reply:
That actually sounds preferable, since we need to make sure that filters against a single partition are computed in the same place (to remove the possible duplicates).